virtme/guest/virtme-sshd-script

```diff
@@ -24,18 +24,37 @@ SSH_AUTH_KEYS="${SSH_HOME}/.ssh/authorized_keys"
 if [ "$(stat -f -c "%t" "${SSH_AUTH_KEYS}")" = "${OVERLAYFS}" ]; then
     cat "${SSH_HOME}"/.ssh/id_*.pub >> "${SSH_AUTH_KEYS}" 2>/dev/null
     chown "${virtme_ssh_user}" "${SSH_AUTH_KEYS}" 2>/dev/null
+    chmod 600 "${SSH_HOME}/.ssh/authorized_keys" 2>/dev/null
 fi
 
 # Generate ssh host keys (if they don't exist already).
 CACHE_DIR=${SSH_HOME}/.cache/virtme-ng/.ssh
 mkdir -p "${CACHE_DIR}/etc/ssh"
 ssh-keygen -A -f "${CACHE_DIR}"
+
+# Generate a minimal sshd config.
+SSH_CONFIG=/etc/ssh/sshd_config
+if [ "$(stat -f -c "%t" "${SSH_CONFIG}")" = "${OVERLAYFS}" ]; then
```

Comment:
Why not putting this file elsewhere? e.g. in `/run` or similar? Maybe an issue with the permissions?

Reply:
> Why not putting this file elsewhere? e.g. in `/run` or similar? Maybe an issue with the permissions?

I was also thinking about that and also move authorized_keys in /run as well, so we don't have to deal about potentially overwriting the real authorized_keys on the host.

However, sshd is complaining if I try to move authorized_keys in /run:
```
Authentication refused: bad ownership or modes for directory /run
```

And we don't want to change the permissions of /run otherwise we may break other stuff. But I think it's possible to move sshd_config to /run, will do some tests.

Reply:
Maybe we can move authorized_keys into the cache dir (where we store the host keys), so even if we rewrite authorize_keys on the host it won't be that bad, as it's basically a file generated in the virtme-ng cache.